### PR TITLE
MORE docs for libgit2

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by KristofferC — APPROVED** [review]
I think it would be useful to be a bit more explicit of what the return types of the functions are.

File: base/libgit2/tag.jl [hunk 1]
```diff
@@ -9,11 +14,23 @@ function tag_list(repo::GitRepo)
...
+
+Remove the git tag `tag` from the repository `repo`.
+"""
 function tag_delete(repo::GitRepo, tag::AbstractString)
     @check ccall((:git_tag_delete, :libgit2), Cint,
                   (Ptr{Void}, Cstring, ), repo.ptr, tag)
 end
 
+"""
+    LibGit2.tag_create(repo::GitRepo, tag::AbstractString, commit, msg::AbstractString="", force::Bool=false, sig::Signature=Signature(repo))
```
Comment (tkelman) [inline base/libgit2/tag.jl]: `msg` and on are keyword args

Comment (kshyatt) [inline base/libgit2/tag.jl]: Fixed, thanks!

File: base/libgit2/tag.jl [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
+"""
+    LibGit2.tag_list(repo::GitRepo)
```
Comment (tkelman) [inline base/libgit2/tag.jl]: `-> Vector{String}`

**Comment by kshyatt** [comment]
Passed CI, think I got all the comments, will add things to the other doc PR if not.

**Comment by tkelman** [comment]
the statuses should describe what types they return, not clear what to expect from the current docstrings

**Pull request merged**